control-plane/agents/src/bin/core/controller/wrapper.rs

```diff
@@ -1151,7 +1151,8 @@ impl ClientOps for Arc<tokio::sync::RwLock<NodeWrapper>> {
 
     async fn shutdown_nexus(&self, request: &ShutdownNexus) -> Result<(), SvcError> {
         let dataplane = self.grpc_client_locked(request.id()).await?;
-        if let Some(nexus) = self.read().await.nexus(&request.uuid()) {
+        let nexus = self.read().await.nexus(&request.uuid());
```

Comment:
@abhilashshetty04 you might have done it inadvertently but this test https://github.com/openebs/mayastor-control-plane/blob/b9cbbf86d45f0c867dc4f81773ca4d99c644c08f/tests/bdd/features/ha/node-agent/test_path_replacement.py#L96 it is not actually testing the path replacement, it's just listing the volume?
In fact the current code here on shutdown_nexus is a deadlock, so not sure how you've tested it?

Reply:
Unless one exists already, please raise a ticket to fix those bdd test, thanks.

Reply:
Yes watchdog will report node status as Unknown.